nav2_regulated_pure_pursuit_controller/test/test_regulated_pp.cpp

```diff
@@ -144,6 +150,26 @@ TEST(RegulatedPurePursuitTest, createCarrotMsg)
   EXPECT_EQ(rtn->point.z, 0.01);
 }
 
+TEST(RegulatedPurePursuitTest, findDirectionChange)
```

Comment:
You should also test inputs to it that don't have a reverse and make sure that output is right too (that it doesn't have it)